### ISSUE TITLE
Can not update only the sts_endpoint without blanking out existing access_key and secret_key

### ISSUE THREAD

**Issue by allyunion** [hashicorp/vault#17716]
Is there a way to retain the existing access_key and secret_key credentials and still update the sts_endpoint?

https://github.com/hashicorp/vault/blob/2a1753a469e86434a2e89eec3e103674fc94a02f/builtin/logical/aws/path_config_root.go#L92-L128

Thanks!

**Comment by maxb** [comment]
I don't believe there is, unfortunately. I have found the same misfeature in the AD secrets engine too, though I hadn't got around to opening an issue for that when I saw yours.

Vault is really in need of a systematic separation of config and credentials, especially in every secret engine that implements a rotate-root endpoint. The way it's currently done, it becomes impossible to reconfigure a secret engine once you've used the rotate-root functionality.

**Comment by cipherboy** [comment]
@maxb @allyunion Would adding a PATCH endpoint similar to the ones in PKI work? Vault 1.13 will have a `vault patch` CLI option, so with more server endpoints supporting PATCH capabilities, it could make for a nice release if someone wants to propose a PR some new handlers... :-)
